### ISSUE TITLE
Generating Spectra factors for test set

### ISSUE THREAD

**Issue by mariemoullet** [dpeerlab/spectra#55]
hello, thank you very much for this method and tutorials! 

i was wondering if there was the option to train the spectra model on one dataset and then generate spectra factors / scores for a new dataset? 

following the syntax of the tutorial, something like

```
model = spc.SPECTRA_Model(X = X_train, labels =  np.array(adata_train.obs['cell_type']), 
                          L = L,
                          adj_matrix = adj_matrix,
                          weights = None, lam = 0.01,  delta=0.001,kappa = None, rho = 0.05, 
                          use_cell_types = True)


model.train(X = X_train, labels =  np.array(adata_train.obs['cell_type']),
            num_epochs=2) 

model.test(X = X_test, labels =  np.array(adata_test.obs['cell_type'])) 
```

thank you very much in advance!

**Comment by wallet-maker** [comment]
Great question, this is not implemented at the moment. 

If the genes are identical in training in test it should be possible to do this by taking the model object output and setting `requires_grad=False` in the training.

Let me know if this works. Also, I think it would be really helpful to find a solution here which also works when the gene space is not identical in training and test.
